projects/soft_999_eddy_kim_fw_103/src/main.c

```diff
@@ -0,0 +1,49 @@
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+static void callback(const GpioAddress *address, void *context) {
+  // Is it right to pass the GpioAddress A6 by *context? I'm unsure what *address is supposed to
+  // return
+  GpioAddress *A6 = context;
+  uint16_t A6_data = 0;
+  adc_read_raw_pin(*A6, &A6_data);
+  LOG_DEBUG("%d\n", A6_data);
+  return;
+}
+
+int main(void) {
+  gpio_init();
+  interrupt_init();
+
+  // Unsure if these are defined properly
+  GpioAddress B2 = { .port = GPIO_PORT_B, .pin = 2 };
+  GpioAddress A6 = { .port = GPIO_PORT_A, .pin = 6 };
```

Comment:
These are defined properly :) Though "B2" and "A6" just refer to the port/pin combination, not the variable name - in fact you should name them something that reflects their purpose, like e.g. `button_pin`, instead of just what the pin is.